### PR TITLE
:tada: Source Snapchat Marketing - enable default availability strategy

### LINKED ISSUE #21787: Source Snapchat Marketing: Enable AvailabilityStrategy

### PR DESCRIPTION
**Pull request by midavadim**
## What
*Describe what the change is solving*
*It helps to add screenshots if it affects the frontend.*

## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

### CODE REVIEW

**Review by lazebnyi — APPROVED** [review]

**Review by erohmensing — APPROVED** [review]
Looks good - I added a commit removing custom error handling that shouldn't be necessary anymore. 

In future PRs, if you could check against [this doc ](https://docs.google.com/document/d/1ghkeIhlfqx7yf6fi3sv3nbpEk7BUpIrVlPqkBmrXR0U/edit) for anything to remove when implementing availabilitystrategy, that'd be great! I also added a note to the issue for each one that has an entry in that list.

**Comment by midavadim** [comment]
@erohmensing 
I have added a comment about removed 'should_retry'

File: airbyte-integrations/connectors/source-snapchat-marketing/source_snapchat_marketing/source.py [hunk 1]
```diff
@@ -219,13 +214,6 @@ def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapp
                 raise SnapchatMarketingException(error_text)
             yield resp.get(self.response_item_name)
 
-    def should_retry(self, response: requests.Response) -> bool:
```
[the remark this answers is not in the document]
Comment (erohmensing) [reply airbyte-integrations/connectors/source-snapchat-marketing/source_snapchat_marketing/source.py:222]: Do you know why it was failing for one specific slice? That would indeed not be covered by the availability strategy, but I'm curious as to why it happened

**Comment by erohmensing** [comment]
@midavadim thanks, responded! Do you know why that was the case? If its legit, I am fine leaving the error handling for now, but I wouldn't expect to run into that for one stream slice so am wondering how the case occurred

File: airbyte-integrations/connectors/source-snapchat-marketing/source_snapchat_marketing/source.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (midavadim) [reply airbyte-integrations/connectors/source-snapchat-marketing/source_snapchat_marketing/source.py:222]: @erohmensing 

related PR: https://github.com/airbytehq/airbyte/pull/20865
related issue: https://github.com/airbytehq/oncall/issues/1067

The problem happened with 'segments' stream for one of 'adaccount' in its slice.

As far as I understand snapchat credentials could have access to a couple of snapchat organization/adaccounts and for one of such accounts there was no permission access to 'segments' data.

**Comment by midavadim** [comment]
> @midavadim thanks, responded! Do you know why that was the case? If its legit, I am fine leaving the error handling for now, but I wouldn't expect to run into that for one stream slice so am wondering how the case occurred

@erohmensing 

related PR: https://github.com/airbytehq/airbyte/pull/20865
related issue: https://github.com/airbytehq/oncall/issues/1067

The problem happened with 'segments' stream for one of 'adaccount' in its slice.

As far as I understand snapchat credentials could have access to a couple of snapchat organization/adaccounts and for one of such accounts there was no permission access to 'segments' data.

**Comment by erohmensing** [comment]
Looking at that issue and the logs for the broken connection, I don't see any indication why it would only be breaking for that one slice, just that that slice was the first one in the list. (I don't see any indication for example that they had access to the segments stream for another adaccount for example) 

Since this is extra error handling, it doesn't hurt us to have it in, but I don't believe it's doing anything. If you feel strongly about it, we can leave it in for now and try removing it later on its own after comparing log messages to see if both cases still happen independently of each other.

File: airbyte-integrations/connectors/source-snapchat-marketing/source_snapchat_marketing/source.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (erohmensing) [reply airbyte-integrations/connectors/source-snapchat-marketing/source_snapchat_marketing/source.py:222]: Will leave this up to you as per my comment in the PR itself!

**Review by erohmensing — APPROVED** [review]

**Pull request merged**